### ISSUE TITLE
MissingMethodException when calling AmqpFileNotificationReceiver.ReceiveAsync() from UWP

### ISSUE THREAD

**Issue by wickste** [Azure/azure-iot-sdk-csharp#610]
- **OS, version, SKU and CPU architecture used:** Windows10 Enterprise (1803) x64 
- **Application's .NET Target Framework :** UWP version 17134, Microsoft.NETCore.UniversalWindowsPlatform 6.1.7 
- **Device:** Laptop 
- **SDK version used:** 1.17.0 

## Description of the issue:

MissingMethodException when calling AmqpFileNotificationReceiver.ReceiveAsync() from a UWP app.

The same code works in a .NET Core console app, as well as in a .NET Framework 4.6 console app.

## Code sample exhibiting the issue:

Code taken from here [https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-csharp-csharp-file-upload](url)
```
private async void ReceiveFileUploadNotificationAsync()
{
    serviceClient = ServiceClient.CreateFromConnectionString(connectionString);
    var notificationReceiver = serviceClient.GetFileNotificationReceiver();

    while (true)
    {
        var fileUploadNotification = await notificationReceiver.ReceiveAsync();
        if (fileUploadNotification == null) continue;
        await notificationReceiver.CompleteAsync(fileUploadNotification);
    }
}

```
## Console log of the issue:

System.MissingMethodException
  HResult=0x80131513
  Message=Method not found: 'Void Microsoft.Azure.Amqp.Transport.TlsTransportSettings.set_Certificate(System.Security.Cryptography.X509Certificates.X509Certificate2)'.
  Source=Microsoft.Azure.Devices
  StackTrace:
   at Microsoft.Azure.Devices.AmqpFileNotificationReceiver.<ReceiveAsync>d__15.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at App7.MainPage.<ReceiveFileUploadNotificationAsync>d__4.MoveNext() in C:\Users\swick\source\repos\App7\App7\MainPage.xaml.cs:line 49

**Comment by CIPop** [comment]
@xinchen10 this appears to be a limitation of the AMQP library.

@wickste we've updated our library to use the next version of AMQP library in 1.17.1. Does this repro with the latest version as well?

**Comment by wickste** [comment]
Sorry, not sure where to find 1.17.1. Can you send a pointer?

**Comment by CIPop** [comment]
@wickste My bad: you're using the _Service_ SDK not _Device_ SDK and you're already using the latest. (I need to update the template.)

@xinchen10 : is this a known issue in the AMQP library?

**Comment by xinchen10** [comment]
The UWP lib is on target platform version 10.0.14393 which does not have many types under System.Security.Cryptography. As a result the Certificate property is not available on TlsTransportSettings. We can move on to later versions but that means the old ones will not be supported any more.

**Comment by wickste** [comment]
@xinchen10 It seems reasonable to move to 16299 at this point, which enables .NET Standard 2.0. Apps that need to target older versions of Windows 10 could still do so by sticking with your current version, no?

**Comment by CIPop** [comment]
@wickste I didn't realize that AMQP is not at 16299.

@xinshen55 1703 (15063) is End of service 10/9/2018 for Home&Pro. (https://support.microsoft.com/en-us/help/13853/windows-lifecycle-fact-sheet ; https://www.microsoft.com/en-us/itpro/windows-10/release-information).
If you still need to support Enterprise customers you could create a separate binary for netstandard2.0 which should be resolved for 16299 targets while keeping the old UAP one for backward compat. (For Azure IoT C# SDK, except from some confusion when upgrading caused by VS, we didn't hear any complaints for just dropping old version support.)

I've opened https://github.com/Azure/azure-amqp/issues/120 to track this issue.

**Comment by jenoola** [comment]
Was able to repro the issue. We are blocked here, cannot do anything about it until https://github.com/Azure/azure-amqp/issues/127 and/or https://github.com/Azure/azure-amqp/issues/120 fixed

**Comment by jenoola** [comment]
@wickste please try to add Microsoft.Azure.Amqp 2.4.1 to your project, it should fix the issue.

**Comment by wickste** [comment]
Seems to be working. @jenoola what's the timeline for updating the Microsoft.Azure.Devices.Client nuget to reference Microsoft.Azure.Amqp >= 2.4.1?

**Comment by prmathur-microsoft** [comment]
We are soon going to release new device client which should contain this fix. Eta 3/10

**Comment by wickste** [comment]
@prmathur-microsoft Awesome, thx for the update!

**Comment by prmathur-microsoft** [comment]
We are now trending to 3/25. Will keep you posted

**Comment by jenoola** [comment]
Fix checked in the new release 1.20.0

**Comment by az-iot-builder-01** [comment]
@CIPop, @wickste, @xinchen10, @jenoola, @prmathur-microsoft, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
